File: frontend/src/types/ats.ts
export interface ParsingChecklistItem {
  detected: boolean
  reason: string | null
}

export interface AtsScoreBreakdown {
  overallScore: number
  keywordMatchPct: number
  skillsMatchPct: number
  titleMatchPct: number
  experienceRelevancePct: number
  sectionCompletenessPct: number
  parsingAccuracyPct: number
  formattingCompatibilityPct: number
  readabilityPct: number
  matchedKeywords: string[]
  missingKeywords: string[]
  relatedKeywords: string[]
  formattingWarnings: string[]
  sectionWarnings: string[]
  suggestions: string[]
  parsingChecklist: Record<string, ParsingChecklistItem>
  pageCount: number | null
  onePageOk: boolean
}

export interface QualityGateResult {
  passed: boolean
  failedChecks: string[]
}

export interface ParsedJobDescription {
  jobTitle: string
  requiredSkills: string[]
  preferredSkills: string[]
  technologies: string[]
  yearsOfExperience: string
  educationRequirements: string[]
  certifications: string[]
  responsibilities: string[]
  domainKeywords: string[]
  softSkills: string[]
  tools: string[]
  cloudTechnologies: string[]
  frameworks: string[]
  databases: string[]
}

export interface JobDescription {
  id: number
  company: string | null
  rawText: string
  parsed: ParsedJobDescription
}

export interface BulletChange {
  targetPath: string
  originalText: string
  suggestedText: string
}

export interface SkillAddition {
  category: string
  skill: string
}

export interface TailorResult {
  summary: string
  recommendedSkills: string[]
  skillAdditions: SkillAddition[]
  skillReorder: { category: string; orderedItems: string[] }[]
  keywordMatches: string[]
  missingKeywords: string[]
  bulletChanges: BulletChange[]
  warnings: string[]
  fabricationRisk: boolean
}

export interface AiChange {
  id: number
  targetPath: string
  changeType: string
  originalValue: string
  suggestedValue: string
  status: 'PENDING' | 'ACCEPTED' | 'REJECTED'
  fabricationRisk: boolean
  blockedLockedField: boolean
}

File: backend/app/schemas/ats.py
from pydantic import BaseModel, ConfigDict, Field

from app.schemas.resume import _to_camel


class ParsingChecklistItem(BaseModel):
    model_config = ConfigDict(populate_by_name=True, alias_generator=_to_camel)
    detected: bool
    reason: str | None = None


class AtsScoreBreakdown(BaseModel):
    model_config = ConfigDict(populate_by_name=True, alias_generator=_to_camel)

    overall_score: float
    keyword_match_pct: float
    skills_match_pct: float
    title_match_pct: float
    experience_relevance_pct: float
    section_completeness_pct: float
    parsing_accuracy_pct: float
    formatting_compatibility_pct: float
    readability_pct: float

    matched_keywords: list[str] = Field(default_factory=list)
    missing_keywords: list[str] = Field(default_factory=list)
    related_keywords: list[str] = Field(default_factory=list)

    formatting_warnings: list[str] = Field(default_factory=list)
    section_warnings: list[str] = Field(default_factory=list)
    suggestions: list[str] = Field(default_factory=list)

    parsing_checklist: dict[str, ParsingChecklistItem] = Field(default_factory=dict)
    page_count: int | None = None
    one_page_ok: bool = False


class QualityGateResult(BaseModel):
    model_config = ConfigDict(populate_by_name=True, alias_generator=_to_camel)
    passed: bool
    failed_checks: list[str] = Field(default_factory=list)
